packages/cart/integration-tests/__tests__/services/cart-module/index.spec.ts

```diff
@@ -2332,4 +2333,184 @@ describe("Cart Module Service", () => {
       expect(taxLines?.length).toBe(0)
     })
   })
+
+  describe.only("retrieve with totals decoration", () => {
+    const testSetup = async () => {
+      const [createdCart] = await service.create([
+        {
+          currency_code: "eur",
+        },
+      ])
+
+      await service.addLineItems(createdCart.id, [
+        {
+          quantity: 2,
+          unit_price: 100,
+          title: "test",
+          tax_lines: [
+            {
+              rate: 0.2,
+              code: "TX",
+            },
+          ],
+          adjustments: [
+            {
+              amount: 40,
+              code: "20%",
+            },
+          ],
+        },
+        {
+          quantity: 2,
+          unit_price: 200,
+          title: "test",
+          tax_lines: [
+            {
+              rate: 0.2,
+              code: "TX",
+            },
+          ],
+        },
+      ])
+
+      await service.addShippingMethods(createdCart.id, [
+        {
+          amount: 100,
+          name: "test",
+          tax_lines: [
+            {
+              rate: 0.2,
+              code: "TX",
+            },
+          ],
+          adjustments: [
+            {
+              amount: 20,
+              code: "20%",
+            },
+          ],
+        },
+      ])
+
+      return createdCart
+    }
+
+    const expectedRes = {
+      subtotal: 600,
+
+      total: 784, // 600 (subtotal) + 116 (shipping) + 112 (item tax) + 16 (shipping tax) - 60 (discount)
```

Comment:
q: how do we get 116 for the shipping? wouldn't it be 80 (100-20) and 16 taxes? and the result would be 748